jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpScheme.java

```diff
@@ -87,6 +90,97 @@ public static int normalizePort(String scheme, int port)
         return httpScheme == null ? port : httpScheme.normalizePort(port);
     }
 
+    /**
+     * Normalize a provided scheme, server, and port into a String form.
+     *
+     * @param scheme the scheme, may not be blank or null.
+     * @param server the server hostname, may not be blank or null.
+     * @param port the port
+     * @return the String representation following the general format {@code <scheme>://<server>:<port>}, where
+     * the returned String might not include a port, and will not include the ending {@code /} character.
+     * @see #appendNormalizedUri(StringBuilder, String, String, int)
+     */
+    public static String normalizeUri(String scheme, String server, int port)
+    {
+        return normalizeUri(scheme, server, port, null, null);
+    }
+
+    /**
+     * Normalize a provided scheme, server, and port into a String form.
+     *
+     * @param scheme the scheme, may not be blank or null.
+     * @param server the server hostname, may not be blank or null.
+     * @param port the port
+     * @param path the optional path, may be null
+     * @param query the optional query, may be null (not included in output if path is null)
+     * @return the String representation following the general format {@code <scheme>://<server>:<port>/<path>?<query>}, where
+     * the returned String might not include a port.
+     * @see #appendNormalizedUri(StringBuilder, String, String, int)
+     */
+    public static String normalizeUri(String scheme, String server, int port, String path, String query)
```

Comment:
Ditto re URI.
I would also add `fragment` as this method may be used on the client too.

Reply:
OK, but I hope the client isn't sending the fragment on the HTTP protocol, as that's not spec compliant.